Tests/UnitTests/AdyenCheckout/AdyenCheckoutTests.swift

```diff
@@ -0,0 +1,158 @@
+            storedBcmcDictionary
+        ],
+        "paymentMethods": [
+            creditCardDictionary,
+            blik
+        ]
+    ]
+
+    override func setUp() {
+        super.setUp()
+        mockProvider = AdyenCheckoutProviderMock()
+        configuration = CheckoutConfiguration(context: Dummy.context)
+        paymentMethods = try! AdyenCoder.decode(paymentMethodsDictionary) as PaymentMethods
+    }
+
+    func testSetupWithSession_Success() {
+        let expectedCheckout = AdyenCheckout(
+            configuration: configuration,
+            session: nil,
+            paymentMethods: nil,
+            checkoutAttemptId: "fakeId",
+            presentationDelegate: nil
+        )
+        mockProvider.setupWithSessionResult = .success(expectedCheckout)
+
+        let expectation = expectation(description: "Completion called")
+        AdyenCheckout.setup(
+            with: "sessionId",
+            sessionData: "sessionData",
+            configuration: configuration,
+            presentationDelegate: nil,
+            provider: mockProvider
+        ) { result in
+            if case let .success(checkout) = result {
+                XCTAssertEqual(checkout.checkoutAttemptId, "fakeId")
+            } else {
+                XCTFail("Expected success")
+            }
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 1)
+        XCTAssertTrue(mockProvider.setupSessionCalled)
+    }
+
+    func testSetupWithSession_Failure() {
+        mockProvider.setupWithSessionResult = .failure(TestError())
+        
+        let expectation = expectation(description: "Completion called")
+        AdyenCheckout.setup(
+            with: "sessionId",
+            sessionData: "sessionData",
+            configuration: configuration,
+            presentationDelegate: nil,
+            provider: mockProvider
+        ) { result in
+            if case .failure = result {
+                // Success path for test
+            } else {
+                XCTFail("Expected failure")
+            }
+            expectation.fulfill()
+        }
+        waitForExpectations(timeout: 1)
+    }
+
+    func testSetupWithPaymentMethods_Success() {
+        let expectedCheckout = AdyenCheckout(
+            configuration: configuration,
+            session: nil,
+            paymentMethods: paymentMethods,
+            checkoutAttemptId: "anotherFakeId",
+            presentationDelegate: nil
+        )
+        
+        mockProvider.setupWithPaymentMethodsResult = .success(expectedCheckout)
+        
+        let expectation = expectation(description: "Completion called")
+        AdyenCheckout.setup(
+            with: paymentMethods,
+            configuration: configuration,
+            presentationDelegate: nil,
+            provider: mockProvider
+        ) { result in
+            if case let .success(checkout) = result {
+                XCTAssertEqual(checkout.checkoutAttemptId, "anotherFakeId")
```

Comment:
I think we could squash all such tests into one and check every possible parameter in one test.